### PR TITLE
Fix deadlock and race condition in synchronous Get

### PR DESCRIPTION
**Pull request by kuujo**
This PR fixes a race condition and a potential deadlock in the NB API's handling of `SYNCHRONOUS` `Get`s. I will comment on some of the code changes to explain...

### CODE REVIEW

File: pkg/northbound/gnmi/v2/get.go
Comment (kuujo) [inline pkg/northbound/gnmi/v2/get.go:185]: Because of the way Go's `for` loops work, we have to pass the target info into the goroutine, otherwise the `target.configuration.Status.Committed.Index` used in the condition above will basically just use the last target in the loop. I doubt this would have worked with multiple targets.

File: pkg/northbound/gnmi/v2/get.go
Comment (kuujo) [inline pkg/northbound/gnmi/v2/get.go:199]: We have to account for the scenario in which the client canceled the context while waiting for the configuration to be propagated, e.g. due to a timeout.

File: pkg/northbound/gnmi/v2/get.go
Comment (adibrastegarnia) [reply pkg/northbound/gnmi/v2/get.go:185]: yep good catch.

**Comment by kuujo** [comment]
I'm running the integration tests. Once they all pass I think we can merge this when you're ready @adibrastegarnia

**Review by adibrastegarnia — APPROVED** [review]

**Comment by kuujo** [comment]
passed

**Pull request merged**
